### DISCUSSION TITLE
Edit HTML - add code to headers and embed div

### DISCUSSION THREAD

**Discussion by phill579** [GitbookIO/community#199] [Feature Requests]
### What's your feature/idea?

I want to be able to have more flexibility for the email marketing options other than Mailchimp integration.

### What's your desired outcome?

I've been using Sender.net and want to embed the subscription form into my gitbook but i'd need to add some javascript to the HTML head and embed the form using some div html

### What's the impact of this for you?

- [X] Critical
- [ ] High
- [ ] Medium
- [ ] Low

**Comment by addisonschultz** [comment]
Hey @phill579 

Great idea. This is definitely something that can already be built with our Integration Platform! We're wrapping up a few things on our end, but it's already fully functional. You can find the docs for it here: https://developer.gitbook.com/

I've also added your request into our backlog for us to visit in the future!

**Comment by ethanlee16** [comment]
@addisonschultz If I'm not mistaken, this is not possible because the scope required to inject JS is restricted to Gitbook-authored integrations only: https://developer.gitbook.com/integrations/configurations#scopes

I am trying to accomplish something similar, for HubSpot. Any way I can accomplish this with the Integration Platform?

**Reply by addisonschultz** [reply]
Hi @ethanlee16, that's correct - this scope is needed in order to make an integration like this, but it's unfortunately still only available for GitBook employees - 

What integration are you building exactly? Is it a requirement to embed js in the page's header in order for it to work?

**Reply by ethanlee16** [reply]
Hi @addisonschultz, it is a requirement for HubSpot's attribution tracking, see here: https://knowledge.hubspot.com/reports/install-the-hubspot-tracking-code

This would also be a requirement for @phill579's sender.net request above ([docs](https://help.sender.net/knowledgebase/embedded-form/)).

**Reply by ethanlee16** [reply]
Is this something we can get access to, to build out the HubSpot integration specifically? The code is virtually identical to the existing Google Analytics and Intercom integrations (inject a script with a configurable ID). We also already have the CSP details for HubSpot to use for the `contentSecurityPolicy` key of the config.

**Reply by addisonschultz** [reply]
Hi Ethan, 

Unfortunately the scope is still not available to external developers - I do understand though the usefulness of adding it in, but we're not there yet in terms of opening it up. I'll make sure to update here once we make more progress on this front though

**Reply by addisonschultz** [reply]
Hi @ethanlee16 - I wanted to send a quick update on this: We've worked with Tabnine to create a Hubspot integration 😄 

You can already install it into your own team. Using the following link format:

https://app.gitbook.com/o/<organization_id>/integrations/hubspot

You can find your organization ID in the URL when working in GitBook, behind the /o/ identifier in the URL. More info on that here: https://developer.gitbook.com/getting-started/concepts#less-than-organizationid-greater-than

Hope this helps!

**Comment by lukewakefield** [comment]
Struggling to get an answer on this so hopefully someone can help. We just want to embed our Live Chat button (we use getgist.com) on our Gitbook docs...it's just JS code, do we have to build an integration just to do that? Is that even possible or are only approved integrations allowed to embed JS into the HTML? It's a very common and simple feature to have a Live Chat button show on docs surely? Thanks.

**Reply by addisonschultz** [reply]
Hey @lukewakefield - that's basically correct. In order to add a service like the Gist to your published docs, you'll need to add Gist's JS to the page - which is only possible to do in GitBook via an integration. For context, adding JS to a published doc site could potentially be malicious, so we limit the use of this through partnered/verified integrations. 

If this is something you would use - I'd be happy to work with you to get this up and running for you, your team, and potentially other GitBook users that also use this tool.

Feel free to reach out to me via addison (@) gitbook.com to discuss this a bit more!

**Reply by lukewakefield** [reply]
Thanks for the quick reply @addisonschultz! Makes sense, that would be really helpful, I'll drop you an email and loop in a team member too. I imagine other Gist users might use Gitbook so would be great to make it a public integration. Cheers!

**Comment by Pascal-Lohscheidt** [comment]
Hey,
I hope this is not to much off topic.
we have a [product](https://www.stepsailor.com/) that offers an assistant to search through your documentation and other features around knowledge bases.
So I would like to integrate gitbook in a few different ways.

One of them would be making the assistant visible in our customers gitbook sites.
But this would require the permission to inject JS 

If I understood correctly it requires a verification or Partnership.
Can I write someone for that? Is this a use case that you do not support anymore? 

Thanks for your help :)

**Reply by addisonschultz** [reply]
Hi @Pascal-Lohscheidt - can you reach out to me at addison @ gitbook.com? Happy to explore this with you!

**Reply by Pascal-Lohscheidt** [reply]
@addisonschultz thanks for the reply I reached out via email
